firedrake/linear_solver.py

```diff
@@ -1,34 +1,24 @@
-from firedrake.exceptions import ConvergenceError
-import firedrake.function as function
-import firedrake.cofunction as cofunction
-import firedrake.vector as vector
-import firedrake.matrix as matrix
-import firedrake.solving_utils as solving_utils
-from firedrake import dmhooks
-from firedrake.petsc import PETSc, OptionsManager, flatten_parameters
-from firedrake.utils import cached_property
-from firedrake.ufl_expr import action
+from firedrake.function import Function
+from firedrake.cofunction import Cofunction
+from firedrake.matrix import MatrixBase
+from firedrake.petsc import PETSc
 from pyop2.mpi import internal_comm
+from firedrake.variational_solver import LinearVariationalProblem, LinearVariationalSolver
 
 __all__ = ["LinearSolver"]
 
 
-class LinearSolver(OptionsManager):
-
-    DEFAULT_KSP_PARAMETERS = solving_utils.DEFAULT_KSP_PARAMETERS
+class LinearSolver(LinearVariationalSolver):
 
     @PETSc.Log.EventDecorator()
-    def __init__(self, A, *, P=None, solver_parameters=None,
-                 nullspace=None, transpose_nullspace=None,
-                 near_nullspace=None, options_prefix=None,
-                 pre_apply_bcs=True):
+    def __init__(self, A, *, P=None, **kwargs):
         """A linear solver for assembled systems (Ax = b).
```

Comment:
I think it would be useful to explicitly state that the matrix cannot change (`constant_jacobian=True`), because this is a different default from `LinearVariationalSolver` (even if it doesn't change the default for `LinearSolver`).
```suggestion
        """A linear solver for assembled systems (Ax = b) with constant A.
```